SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRemotePossession.cpp

```diff
@@ -25,31 +26,87 @@ void ASpatialTestRemotePossession::PrepareTest()
 {
 	Super::PrepareTest();
 
-	AddStep(TEXT("EnsureSpatialOS"), FWorkerDefinition::AllServers, nullptr, [this]() {
+	AddStep(TEXT("EnsureSpatialOS"), FWorkerDefinition::AllServers, nullptr, /*StartEvent*/ [this]() {
 		ULayeredLBStrategy* LoadBalanceStrategy = GetLoadBalancingStrategy();
 		AssertTrue(LoadBalanceStrategy != nullptr, TEXT("Test requires SpatialOS enabled with Load-Balancing Strategy"));
 		FinishStep();
 	});
 
-	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float DeltaTime) {
+	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, /*StartEvent*/ [this]() {
 		ATestPossessionPawn* Pawn =
 			GetWorld()->SpawnActor<ATestPossessionPawn>(LocationOfPawn, FRotator::ZeroRotator, FActorSpawnParameters());
 		RegisterAutoDestroyActor(Pawn);
 		FinishStep();
 	});
 
+	AddStep(TEXT("Clear Original Pawn Array"), FWorkerDefinition::AllServers, nullptr, /*StartEvent*/ [this]() {
+		OriginalPawns.Empty();
+		FinishStep();
+	});
+
 	// Ensure that all Controllers are located on the right Worker
 	AddWaitStep(FWorkerDefinition::AllServers);
 
+	AddStep(TEXT("Save Original Pawns"), FWorkerDefinition::AllServers, nullptr, /*StartEvent*/ [this]() {
+		for (ASpatialFunctionalTestFlowController* FlowController : GetFlowControllers())
+		{
+			ATestPossessionPlayerController* PlayerController = Cast<ATestPossessionPlayerController>(FlowController->GetOwner());
+			if (PlayerController != nullptr && PlayerController->HasAuthority() && PlayerController->GetPawn() != nullptr
```

Comment:
Maybe we could be slightly more strict here, e.g. asserting that the pawn is not null and that we have authority over it IF we have authority over the playerController.